sdk/resourcemanager/azure-resourcemanager-compute/src/main/java/com/azure/resourcemanager/compute/implementation/VirtualMachineImpl.java

```diff
@@ -2699,17 +2699,37 @@ public VirtualMachineImpl withExistingVirtualMachineScaleSet(VirtualMachineScale
     }
 
     @Override
-    public VirtualMachineImpl swapOSDisk(String diskId) {
-        if (diskId == null
-            || !isManagedDiskEnabled()
-            || this.innerModel().storageProfile().osDisk().managedDisk() == null) {
+    public VirtualMachineImpl withOSDisk(String diskId) {
+        if (diskId == null) {
             return this;
         }
-        this.storageProfile().osDisk().withName(ResourceUtils.nameFromResourceId(diskId));
+        return withOSDisk(diskId, ResourceUtils.nameFromResourceId(diskId));
+    }
+
+    @Override
+    public VirtualMachineImpl withOSDisk(String diskId, String diskName) {
```

Comment:
what is the effect of a new `diskName`?

Reply:
I just checked that `diskName` must match the resource ID... So this overload is useless, will remove it.